### ISSUE TITLE
[FEATURE] Docs - Create an example using the new flat config and extend function

### ISSUE THREAD

**Issue by shoulders** [yeonjuan/html-eslint#374]
### The issue
I am a newbie to NodeJS, ESLint and HTML-ESLINT and I found it very difficult to get my head around this new `Flat Config`. It has changed so much recently, there are no examples on how to set this up correctly. My IDE of choice is VSCode.

I looked at the examples on this page but the did not related to this new format.

https://html-eslint.org/docs/getting-started

There is a section called `Flat config` but it has nothing in it.

### The Request

Can you update your examples to reflect the new format that utilises the `extend` function.

### Example code that works with this new format

Using a default file I generated with `npx eslint --init` I added the following code

**Add this import statement (just above import { defineConfig } from "eslint/config");**
```javascript
import html from "@html-eslint/eslint-plugin";
```

**Add a files: declaration at the end of the others. Pick one from below as they both do the same (see their comments)**
```javascript
// This matches the other rule's format for easy of use.
{ files: ["**/*.html"], plugins: { html }, language: "html/html", extends: [html.configs["flat/recommended"]] },
 
or
 
// This version is the same as above with the addtion of an example rule override and a different layout to make adding more rule overrides easier.
{
  files: ["**/*.html"],
  plugins: {
    html,
  },
  language: "html/html",
  extends: [html.configs["flat/recommended"]],
  rules: {
    "html/no-duplicate-class": "error",
  }
},
```

#### The complete file eslint.config.mjs

```javascript
import js from "@eslint/js";
import globals from "globals";
import json from "@eslint/json";
import markdown from "@eslint/markdown";
import css from "@eslint/css";
import html from "@html-eslint/eslint-plugin";
import { defineConfig } from "eslint/config";


export default defineConfig([
  { files: ["**/*.{js,mjs,cjs}"], plugins: { js }, extends: ["js/recommended"] },
  { files: ["**/*.{js,mjs,cjs}"], languageOptions: { globals: globals.browser } },
  { files: ["**/*.json"], plugins: { json }, language: "json/json", extends: ["json/recommended"] },
  { files: ["**/*.jsonc"], plugins: { json }, language: "json/jsonc", extends: ["json/recommended"] },
  { files: ["**/*.json5"], plugins: { json }, language: "json/json5", extends: ["json/recommended"] },
  { files: ["**/*.md"], plugins: { markdown }, language: "markdown/gfm", extends: ["markdown/recommended"] },
  { files: ["**/*.css"], plugins: { css }, language: "css/css", extends: ["css/recommended"] },
  { files: ["**/*.html"], plugins: { html }, language: "html/html", extends: [html.configs["flat/recommended"]] },
]);
```

### Research

I am posting the link to may article to show I have Googled and done my research. Maybe this issue or my article might help someone in the same boat.

https://quantumwarp.com/kb/articles/26-programming/1032-my-visual-studio-code-notes

**Comment by yeonjuan** [comment]
@shoulders Thanks for the suggestion! I agree that the documentation needs an update.
